src/core/ext/lb_policy/grpclb/grpclb.c

```diff
@@ -804,8 +805,11 @@ static char *get_lb_uri_target_addresses(grpc_exec_ctx *exec_ctx,
       char *addr_str;
       GPR_ASSERT(grpc_sockaddr_to_string(
                      &addr_str, &addresses->addresses[i].address, true) > 0);
-      targets_info_entries[addr_index] = targets_info_entry_create(
-          addr_str, addresses->addresses[i].balancer_name);
+      if (addresses->addresses[i].balancer_name != NULL) {
+        targets_info_entries[addr_index] = targets_info_entry_create(
```

Comment:
If we do actually have to handle this case for some reason, shouldn't this array index be `targets_info_entries_size` instead of `addr_index`?